From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Fri, 28 May 2021 03:28:15 -0500
Subject: [PATCH] Revert EntityPickupItemEvent for villagers - SPIGOT-6452

https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/8a516e0bbb025dc531a749a3b2fdbbc4359d89c8
Cancelling is improperly implemented as it is updated into the villager's inventory and the item stays on the ground.
Reverting the commit till upstream implements a fix for this.
---
 .../minecraft/world/entity/npc/EntityVillager.java | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/main/java/net/minecraft/world/entity/npc/EntityVillager.java b/src/main/java/net/minecraft/world/entity/npc/EntityVillager.java
index 5e0565ef9d1f34b0a758aeec109202b5a1b68d7c..6ea00fe75c67700784ab9c5fbaeb7134151b596e 100644
--- a/src/main/java/net/minecraft/world/entity/npc/EntityVillager.java
+++ b/src/main/java/net/minecraft/world/entity/npc/EntityVillager.java
@@ -843,16 +843,16 @@ public class EntityVillager extends EntityVillagerAbstract implements Reputation
                 return;
             }
 
-            // CraftBukkit start
-            ItemStack itemstack1 = inventorysubcontainer.a(itemstack);
-            if (CraftEventFactory.callEntityPickupItemEvent(this, entityitem, itemstack1.getCount(), false).isCancelled()) {
-                return;
-            }
-            // CraftBukkit end
+            // CraftBukkit start // EMC start - revert till fixed
+            //ItemStack itemstack1 = inventorysubcontainer.a(itemstack);
+            //if (CraftEventFactory.callEntityPickupItemEvent(this, entityitem, itemstack1.getCount(), false).isCancelled()) {
+            //    return;
+            //}
+            // CraftBukkit end // EMC end - revert till fixed
 
             this.a(entityitem);
             this.receive(entityitem, itemstack.getCount());
-            // ItemStack itemstack1 = inventorysubcontainer.a(itemstack); // CraftBukkit - moved up
+            ItemStack itemstack1 = inventorysubcontainer.a(itemstack); // CraftBukkit - moved up // EMC - revert
 
             if (itemstack1.isEmpty()) {
                 entityitem.die();
